lsp/source/server.mts

```diff
@@ -501,6 +514,11 @@ async function executeQueue() {
   // Run all jobs in queue (preventing livelock).
   for (const document of changed) {
     await updateDiagnosticsForDoc(document)
+    documentUpdateStatus.get(document.uri)?.resolve(true)
+    Promise
+      .resolve()
```

Comment:
Are these two lines necessary? Or could we apply `.then` to the `.resolve(true)`?